crates/sui-core/src/consensus_handler.rs

```diff
@@ -200,6 +200,14 @@ fn update_index_and_hash(
     true
 }
 
+impl<C> Drop for ConsensusHandler<C> {
+    fn drop(&mut self) {
+        // Assuming ConsensusHandler only gets dropped at process shutdown and epoch change,
+        // it is safe to try to release db handles if epoch has advanced.
+        self.epoch_store.release_db_handles();
```

Comment:
oof this is so icky, I wish we didn't need this entire "release tables" thing :/ but not related to your PR, I guess.

Reply:
Why do we need this now? Is there going to be leaks if we don't do it?